linked_list.h

```diff
@@ -102,17 +118,16 @@ int insert_node(void *data, char type, size_t bytes, list **head) {
 }
 
 int insert_char(char data, list ** head) {
-	return insert_node((char *)&data, 'c', sizeof(char), head);
+	return insert_node((char *)&data, 'c', &get_size, head);
 }
 int insert_int(int data, list ** head) {
-	return insert_node((int *)&data, 'i', sizeof(int), head);
+	return insert_node((int *)&data, 'i', &get_size, head);
 }
 int insert_float(float data, list ** head) {
-	return insert_node((float *)&data, 'f', sizeof(float), head);
+	return insert_node((float *)&data, 'f', &get_size, head);

```

Comment:
Change this to
insert_node((void *)&data, &getFloatSize,&printFloat, head);

Reply:
changed

Reply:
Remove type information altogether, Think how you can do it. Why do you want to send 'f' anyways? 
You are sending 'f' to insert_node, which in turn will send it to get_size, which in turn it will call sizeof(float) , 

Instead of all this, you can write a new getFloatSize method, and pass its address here only right? removing type altogether.